src/persist-client/src/read.rs

```diff
@@ -517,6 +630,13 @@ where
         Ok(Listen::new(self, as_of).await)
     }
 
+    /// Returns a [`BatchFetcher`], which is only capable of fetching batches.
+    #[instrument(level = "debug", skip_all, fields(shard = %self.machine.shard_id()))]
+    pub fn batch_fetcher(self) -> BatchFetcher<K, V, T, D> {
```

Comment:
This one seems tricky:
 - I think the `BatchFetcher` should not hold on to a `since` or a `SeqNo`. All of that is normally being taken care of by the `ReadHandle` that emits batches.
 - It should also not be reflected in state.
 - It seems all the fetcher really needs is a handle to `Blob`, a handle to metrics, and it's `ShardId` (for sanity checking).
 - This would also mean that the fetcher does not do heartbeats anymore and relies on someone doing that on the `ReadHandle`. Is that fine?

Right now, creating a `BatchFetcher` consumes the `ReadHandle`, meaning that we have to `clone()` the `ReadHandle` in `Listen`, because it want's to hold on to both. This means we would temporarily create and entry in state and then clean it up again when the clone is consumed to create the `BatchFetcher`. Alternatively, we could add an `open_fetcher()` to `PersistClient` itself. 

cc @danhhz because some of these might be a bit out there.

Reply:
For this PR, we should do the most straightforward thing: `batch_fetcher` takes the things it needs out of `ReadHandle` and `expire()`s it. And then we can think about more esoteric changes.

Reply:
> I think the BatchFetcher should not hold on to a since or a SeqNo. All of that is normally being taken care of by the ReadHandle that emits batches.

good point. this BatchFetcher wrapping ReadHandle pattern I think came from one of my comments on the last PR, but that comment has been obsoleted by this new SeqNo leasing stuff y'all came up with. I agree that with the new approach, BatchFetcher should be "dumb".

> For this PR, we should do the most straightforward thing: batch_fetcher takes the things it needs out of ReadHandle and expire()s it. And then we can think about more esoteric changes.
wfm!

Reply:
Done in commit w/ message `persist: refactor BatchFetcher to contain only necessary parts of ReadHandle, not own it`